### PR TITLE
Defoonder

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request
Defoonder
## Why It's Good For The Game

image

## Changelog
:cl:
add: Defoonder
/:cl:

### CODE REVIEW

**Comment by token-map** [comment]
speedmerge

**Comment by UntoldTactics** [comment]
YESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYESYES

**Comment by lKiseki** [comment]
speedmerge
right now

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm [hunk 1]
```diff
@@ -144,3 +144,151 @@
 	crest_defense_armor = 30
 	crest_defense_slowdown = 0.8
 	fortify_armor = 55
+
+/datum/xeno_caste/defoonder
+	caste_name = "Defoonder"
+	display_name = "Defoonder"
+	upgrade_name = ""
+	caste_desc = "He attac."
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:152]:
```suggestion
	caste_desc = "He deffen"
```

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm [hunk 2]
```diff
@@ -144,3 +144,151 @@
...
+
+
+/datum/xeno_caste/defoonder/young
+	upgrade_name = "Young"
+
+	upgrade = XENO_UPGRADE_ZERO
+
+/datum/xeno_caste/defoonder/mature
+	upgrade_name = "Mature"
+	caste_desc = "He attac. He smac."
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:213]:
```suggestion
	caste_desc = "He deffen He protecc."
```

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm [hunk 3]
```diff
@@ -144,3 +144,151 @@
...
+	soft_armor = list("melee" = 33, "bullet" = 33, "laser" = 28, "energy" = 25, "bomb" = XENO_BOMB_RESIST_2, "bio" = 25, "rad" = 25, "fire" = 33, "acid" = 25)
+
+	// *** Defender Abilities *** //
+	crest_defense_armor = 26
+	crest_defense_slowdown = 0.8
+	fortify_armor = 52
+
+/datum/xeno_caste/defoonder/elder
+	upgrade_name = "Elder"
+	caste_desc = "He attac. He smac. He snac."
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:239]:
```suggestion
	caste_desc = "He deffen. He protecc. He fuwutify."
```

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm [hunk 4]
```diff
@@ -144,3 +144,151 @@
...
+	soft_armor = list("melee" = 35, "bullet" = 35, "laser" = 30, "energy" = 28, "bomb" = XENO_BOMB_RESIST_2, "bio" = 28, "rad" = 28, "fire" = 35, "acid" = 28)
+
+	// *** Defender Abilities *** //
+	crest_defense_armor = 30
+	crest_defense_slowdown = 0.8
+	fortify_armor = 55
+
+/datum/xeno_caste/defoonder/ancient
+	upgrade_name = "Ancient"
+	caste_desc = "He attac. He smac. He snac. But most importantly, He PROTEC!"
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:268]:
```suggestion
	caste_desc = "He deffen. He protecc. He fuwutify. But most importantly, He THICC!"
```

**Comment by Ragecrabbo** [comment]
NEED

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm (hunk 4 above)
Comment (Alexeivichi) [reply code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:268]: "But most importantly, He THICC!"
What in the f...?

**Comment by Oscari-pog** [comment]
blessed

**Comment by Kuro020** [comment]
Wtf

**Review by Kuro020 — APPROVED** [review]

**Comment by Mrrpip** [comment]
Fren

**Comment by DenialDuckOfService** [comment]
Another blessed change

File: code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm (hunk 4 above)
Comment (UntoldTactics) [reply code/modules/mob/living/carbon/xenomorph/castes/defender/castedatum_defender.dm:268]: Excellent

**Review by BraveMole — CHANGES_REQUESTED** [review]
If you intend to make a roony version for each xeno, you need to improve the roony code.

This should only be a sprite change (and maybe name + desc change), you should not have to add defoonder/bump for exemple.

Make it so sprites have a variant if a var is TRUE of the xenomorph, and fix roony code as well

**Review by BraveMole — APPROVED** [review]

**Pull request merged**